analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/LoopDetector.cs

```diff
@@ -83,11 +83,22 @@ void MergeWithIntersectingLoops(int[] path, int last)
         }
     }
 
-    private static IEnumerable<int> Successors(ControlFlowGraph cfg, int block)
+    private static IEnumerable<int> Successors(ControlFlowGraph cfg, int index)
     {
-        var successors = cfg.Blocks[block].SuccessorBlocks.Select(x => x.Ordinal);
-        return cfg.Blocks[block].EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion
-            ? successors.Concat(tryRegion.ReachableHandlers().Select(x => x.FirstBlockOrdinal))
-            : successors;
+        var block = cfg.Blocks[index];
+        var successors = block.SuccessorBlocks.Select(x => x.Ordinal);
+        if (block.EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion)
+        {
+            successors = successors.Concat(tryRegion.ReachableHandlers().Select(x => x.FirstBlockOrdinal));
+        }
+        if (block.EnclosingRegion(ControlFlowRegionKind.Finally) is { } finallyRegion)
+        {
+            successors = successors.Concat(
+                cfg.Blocks.Where(x => x.IsIn(finallyRegion.EnclosingRegion.NestedRegion(ControlFlowRegionKind.Try)))
+                    .SelectMany(x => x.SuccessorBlocks)
+                    .Where(x => !x.IsIn(finallyRegion.EnclosingRegion))
```

Comment:
I liked the `tryFinally` extracted in a variable. It made it easier to understand.